pkg/controller/pod.go

```diff
@@ -254,8 +259,13 @@ func (pnc *PodNetworksController) processNextWorkItem() bool {
 				PodSandboxID: podSandboxID,
 			})
 		if err != nil {
+			// The number of results will always less than len of attachmentsToAdd if err != nil.
+			attachmentsToRollback = attachmentsToAdd[:len(results)+1]
 			klog.Errorf("error adding attachments: %v", err)
 			return true
+		} else {
+			// If update pod network status successed we rollbak nothing after handleResult.
```

Comment:
nit: i think we can drop this comment
as it depends on `handleResult`
here we just set `attachmentsToRollback` to equal the whole `attachmentsToAdd` because
`handleDynamicInterfaceRequest` succeed
so the comment might be bit confusing and not required here imho